rust/private/rustc.bzl

```
@@ -509,46 +505,46 @@ def rustc_compile_action(
             # nb. This field is required for cc_library to depend on our output.
             files = depset([crate_info.output]),
             runfiles = runfiles,
-            executable = crate_info.output if crate_info.type == "bin" or out_binary else None,
+            executable = crate_info.output if crate_info.type == "bin" or "--test" in rust_flags or out_binary else None,
         ),
     ]
 
 def add_edition_flags(args, crate):
     if crate.edition != "2015":
         args.add("--edition={}".format(crate.edition))
 
-def _create_out_dir_action(ctx, file, build_info, dep_info):
+def _create_extra_input_args(ctx, file, build_info, dep_info):
     tar_file_attr = getattr(file, "out_dir_tar", None)
     if build_info and tar_file_attr:
         fail("Target {} has both a build_script dependency and an out_dir_tar - this is not allowed.".format(ctx.label))
 
-    prep_commands = []
     input_files = []
-    # Env vars and build flags which need to be set in the action's command line, rather than on the action's env,
-    # because they rely on other env vars or commands.
-    dynamic_env = {}
-    dynamic_build_flags = []
 
-    # TODO: Remove system tar usage
+    # Argumentss to the commandline line wrapper that are going to be used
+    # to create the final command line
+    out_dir = None
+    tar_file = None
+    build_env_file = None
```

Typos + deprecating of tar_file
```suggestion
    # Arguments to the command-line wrapper that are going to be used
    # to create the final command line
    out_dir = None
    build_env_file = None
```

---

I feel that removing the Tar file support should be part of another change, wasm_bindgen, and bindgen depend on it and I need to fix them, seemed like I would blow up the change.

---

Indeed you are correct, this is bigger than I expected, let me see if i can create a PR for that.